src/wallet/mod.rs

```diff
@@ -5009,9 +5053,23 @@ pub(crate) mod test {
         ));
 
         // ...Now the coinbase is mature :)
+        let sync_time = SyncTime {
+            block_time: BlockTime {
+                height: maturity_time,
+                timestamp: 0,
+            },
+        };
+        wallet
+            .database
+            .borrow_mut()
+            .set_sync_time(sync_time)
+            .unwrap();
+
+        let balance = wallet.get_balance().unwrap();
+        assert!(balance.immature == 0 && balance.confirmed != 0);
```

Comment:
Following evan's suggestion, I think that here we should assert_eq on the whole balance structure, just to make sure that if we have some weird inconsistencies they don't go unnoticed